promql/engine.go

```diff
@@ -379,6 +436,36 @@ func (ng *Engine) exec(ctx context.Context, q *query) (Value, storage.Warnings,
 	ctx, cancel := context.WithTimeout(ctx, ng.timeout)
 	q.cancel = cancel
 
+	defer func() {
+		ng.queryLoggerLock.RLock()
+		if l := ng.queryLogger; l != nil {
+			f := []interface{}{"query", q.q}
+			switch eq := q.Statement().(type) {
+			case *EvalStmt:
+				f = append(f,
+					"start", formatDate(eq.Start),
+					"end", formatDate(eq.End),
+					"step", eq.Interval.String(),
+				)
+			case testStmt:
+				f = append(f, "test", "yes")
+			default:
+				panic(errors.Errorf("can't log %v", eq))
```

Comment:
@brian-brazil should we panic here ? It might be that third party implementations pass objects that also follow the Query() interface, then the engine would panic for them.

Reply:
Probably best not to, though it'd be the Statement interface which 3rd parties shouldn't need.

But on the other hand why do you log tests? They can only happen in promtool.

Reply:
Removed logging.